.github/workflows/intermittent-test-check.yml

```diff
@@ -87,11 +90,54 @@ jobs:
     with:
       repo: ${{ github.event.inputs.ratis-repo || format('{0}/ratis', github.repository_owner) }}
       ref: ${{ github.event.inputs.ratis-ref }}
+  find-tests:
+    if: ${{ always() }}
+    needs:
+      - prepare-job
+    runs-on: ubuntu-24.04
+    outputs:
+      modules: ${{ steps.modules.outputs.modules }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.ref }}
+      - name: Cache for maven dependencies
+        uses: actions/cache/restore@v4
+        with:
+          path: |
+            ~/.m2/repository/*/*/*
+            !~/.m2/repository/org/apache/ozone
+          key: maven-repo-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            maven-repo-
+      - name: Setup java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: 8
+      - name: Find tests
+        run: |
+          # find tests to be run in splits by running them with very short timeout
+          hadoop-ozone/dev-support/checks/junit.sh -DexcludedGroups="native|slow|unhealthy" -DskipShade \
+              -Dtest="$TEST_CLASS,Abstract*Test*\$*" \
+              -Dsurefire.fork.timeout=1 -Dmaven-surefire-plugin.version=${{ env.SUREFIRE_VERSION }} \
+              || true # ignore errors
+        env:
+          ITERATIONS: 1
+      - name: Find modules
+        id: modules
+        run: |
+          grep -e 'surefire:${{ env.SUREFIRE_VERSION }}:test' -e 'Running org.apache' target/unit/output.log | grep -B1 'Running org.apache'
```

Comment:
Is target/unit/output.log populated even before the test run?

Reply:
`Find tests` actually executes the tests once, with timeout=1s.  I didn't find any other way to reliably find which tests Surefire/JUnit would run.

Reply:
yeah I checked too but couldn't find better. Even if it is not able to write within 1 sec due to any issue, it will fallback to building all modules which is what happens today so I guess it should be okay.